kernel/memory_structs/src/lib.rs

```diff
@@ -470,6 +486,16 @@ macro_rules! implement_page_frame_range {
                         None
                     }
                 }
+
+                #[doc = "Returns true if `other` is contained within the `" $TypeName "`."]
+                pub fn contains_range(&self, other: &$TypeName) -> bool {
+                    if !other.is_empty() && (other.start() >= self.start()) && (other.end() <= self.end()) 
+                    {
+                        true
+                    } else {
+                        false
+                    }
```

Comment:
surprised that clippy didn't catch this, but you can just return the boolean conditional directly instead of doing `if (...) { true } else { false }`